.github/workflows/require_be_approval.yml

```diff
@@ -63,39 +154,105 @@ jobs:
           GITHUB_TOKEN: ${{ env.VA_VSP_BOT_GITHUB_TOKEN }}
 
       - name: Get PR reviews
-        if: steps.approval_permissions.outputs.require_be_approval == 'true'
         id: get_pr_reviews
         uses: octokit/request-action@v2.x
         with:
-          route: GET /repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number || github.event.pull_request_review.pull_request.number }}/reviews
+          route: GET /repos/${{ github.repository }}/pulls/${{ env.pr_number }}/reviews
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Verify backend-review-group approval
-        if: steps.approval_permissions.outputs.require_be_approval == 'true'
         id: verify_approval
         run: |
           BACKEND_REVIEWERS=$(cat <<'EOF' | jq -r '.[].login' | tr '\n' '|' | sed 's/|$//'
           ${{ steps.get_team_members.outputs.data }}
           EOF
           )
 
-          APPROVALS=$(cat <<'EOF' | jq -r '.[] | select(.state == "APPROVED") | .user.login' | grep -iE "$BACKEND_REVIEWERS" | wc -l
+          # Get PR author
+          PR_AUTHOR=$(gh pr view ${{ env.pr_number }} --json author -q '.author.login')
+
+          # Check if PR author is in BACKEND_REVIEWERS
+          if echo "$PR_AUTHOR" | grep -qiE "^($BACKEND_REVIEWERS)$"; then
+            echo "PR author '$PR_AUTHOR' is a backend-review-group member."
+            echo "author_is_backend=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "PR author '$PR_AUTHOR' is NOT a backend-review-group member."
+            echo "author_is_backend=false" >> "$GITHUB_OUTPUT"
+          fi
+
+          TEAM_APPROVALS=$(cat <<'EOF' | jq -r '[.[] | {user: .user.login, state, submitted_at}]
+            | sort_by(.submitted_at)
+            | reverse
+            | unique_by(.user)
+            | map(select(.state == "APPROVED"))
+            | .[].user' | grep -viE "$BACKEND_REVIEWERS" | wc -l
+          ${{ steps.get_pr_reviews.outputs.data }}
+          EOF
+          )
+
+          echo "Number of non backend-review-group approvals: $TEAM_APPROVALS"
+          if [ "$TEAM_APPROVALS" -eq 0 ]; then
+            echo "team_approval_status=required" >> "$GITHUB_OUTPUT"
+          else
+            echo "team_approval_status=confirmed" >> "$GITHUB_OUTPUT"
+          fi
+
+          BE_APPROVALS=$(cat <<'EOF' | jq -r '[.[] | {user: .user.login, state, submitted_at}]
+            | sort_by(.submitted_at)
+            | reverse
+            | unique_by(.user)
+            | map(select(.state == "APPROVED"))
+            | .[].user' | grep -iE "$BACKEND_REVIEWERS" | wc -l
           ${{ steps.get_pr_reviews.outputs.data }}
           EOF
           )
 
```

Comment:
Only the latest reviewers will be fetched when pulling TEAM_APPROVALS and BE_APPROVALS